
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.6.3">
    
    
      
        <title>Single Nucleotide Variant Calling and Annotation - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-56ade86843.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-e2807e330f.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-f78e5cb881.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/spoilers.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-logo md-header-nav__button">
            <img src="../../../media/logos/bpalogo.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Cancer Genomics Modules
                </span>
              
            
            Single Nucleotide Variant Calling and Annotation
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="BTP Modules" class="md-tabs__link">
          BTP Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../cancer-module-alignment/alignment/" title="Cancer Genomics Modules" class="md-tabs__link md-tabs__link--active">
          Cancer Genomics Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../denovo-module-cli/commandline/" title="DeNovo Modules" class="md-tabs__link">
          DeNovo Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics Modules" class="md-tabs__link">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../../media/logos/bpalogo.png">
      </i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Timetable" class="md-nav__link">
      Timetable
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      BTP Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        BTP Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-denovo-canu/denovo_canu/" title="DeNovo Canu" class="md-nav__link">
      DeNovo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Cancer Genomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Cancer Genomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Single Nucleotide Variant Calling and Annotation
      </label>
    
    <a href="./" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link md-nav__link--active">
      Single Nucleotide Variant Calling and Annotation
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bam-files" title="BAM Files" class="md-nav__link">
    BAM Files
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-exploring-bam-files" title="Step 1: Exploring BAM files" class="md-nav__link">
    Step 1: Exploring BAM files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-pre-processing-indel-realignment" title="Step 2: Pre-processing: Indel Realignment" class="md-nav__link">
    Step 2: Pre-processing: Indel Realignment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-pre-processing-fixmates" title="Step 3: Pre-processing: Fixmates" class="md-nav__link">
    Step 3: Pre-processing: Fixmates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-pre-processing-mark-duplicates" title="Step 4: Pre-processing: Mark Duplicates" class="md-nav__link">
    Step 4: Pre-processing: Mark Duplicates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-pre-processing-base-quality-recalibration" title="Step 4: Pre-processing: Base Quality Recalibration" class="md-nav__link">
    Step 4: Pre-processing: Base Quality Recalibration
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bam-qc" title="BAM QC" class="md-nav__link">
    BAM QC
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-bam-qc-compute-coverage" title="Step 1: BAM QC: Compute Coverage" class="md-nav__link">
    Step 1: BAM QC: Compute Coverage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-bam-qc-insert-size" title="Step 2: BAM QC: Insert Size" class="md-nav__link">
    Step 2: BAM QC: Insert Size
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-bam-qc-alignment-metrics" title="Step 3: BAM QC: Alignment metrics" class="md-nav__link">
    Step 3: BAM QC: Alignment metrics
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-calling" title="Variant Calling" class="md-nav__link">
    Variant Calling
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-visualisation" title="Variant Visualisation" class="md-nav__link">
    Variant Visualisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-annotation" title="Variant Annotation" class="md-nav__link">
    Variant Annotation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-filtration" title="Variant Filtration" class="md-nav__link">
    Variant Filtration
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#acknowledgements" title="Acknowledgements" class="md-nav__link">
    Acknowledgements
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-nav__link">
      Copy Number Variation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-nav__link">
      Variant Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-6" type="checkbox" id="nav-5-6">
    
    <label class="md-nav__link" for="nav-5-6">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-6">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/01_signatures/" title="Assessing somatic mutational signatures" class="md-nav__link">
      Assessing somatic mutational signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/02_intogen/" title="Cohort analysis for the identification of driver genes" class="md-nav__link">
      Cohort analysis for the identification of driver genes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      DeNovo Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        DeNovo Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-nav__link">
      DeNovo Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-bac/bac_asm/" title="Bacterial Assembly" class="md-nav__link">
      Bacterial Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-euk/eukaryotic_assem/" title="Eukaryote Assembly" class="md-nav__link">
      Eukaryote Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-vis/vis/" title="Metagenomics Visualisation" class="md-nav__link">
      Metagenomics Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bam-files" title="BAM Files" class="md-nav__link">
    BAM Files
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-exploring-bam-files" title="Step 1: Exploring BAM files" class="md-nav__link">
    Step 1: Exploring BAM files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-pre-processing-indel-realignment" title="Step 2: Pre-processing: Indel Realignment" class="md-nav__link">
    Step 2: Pre-processing: Indel Realignment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-pre-processing-fixmates" title="Step 3: Pre-processing: Fixmates" class="md-nav__link">
    Step 3: Pre-processing: Fixmates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-pre-processing-mark-duplicates" title="Step 4: Pre-processing: Mark Duplicates" class="md-nav__link">
    Step 4: Pre-processing: Mark Duplicates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-pre-processing-base-quality-recalibration" title="Step 4: Pre-processing: Base Quality Recalibration" class="md-nav__link">
    Step 4: Pre-processing: Base Quality Recalibration
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bam-qc" title="BAM QC" class="md-nav__link">
    BAM QC
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-bam-qc-compute-coverage" title="Step 1: BAM QC: Compute Coverage" class="md-nav__link">
    Step 1: BAM QC: Compute Coverage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-bam-qc-insert-size" title="Step 2: BAM QC: Insert Size" class="md-nav__link">
    Step 2: BAM QC: Insert Size
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-bam-qc-alignment-metrics" title="Step 3: BAM QC: Alignment metrics" class="md-nav__link">
    Step 3: BAM QC: Alignment metrics
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-calling" title="Variant Calling" class="md-nav__link">
    Variant Calling
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-visualisation" title="Variant Visualisation" class="md-nav__link">
    Variant Visualisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-annotation" title="Variant Annotation" class="md-nav__link">
    Variant Annotation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-filtration" title="Variant Filtration" class="md-nav__link">
    Variant Filtration
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#acknowledgements" title="Acknowledgements" class="md-nav__link">
    Acknowledgements
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Single Nucleotide Variant Calling and Annotation</h1>
                
                <h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>
<p>Prepare raw BAM alignments for variant detection</p>
</li>
<li>
<p>Perform QC measures on BAM files</p>
</li>
<li>
<p>Understand and perform simple variant detection on paired NGS data</p>
</li>
<li>
<p>Add annotation information to raw variant calls</p>
</li>
<li>
<p>Visualise variant calls using IGV</p>
</li>
</ul>
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>SAMTools:<br />
<a href="http://sourceforge.net/projects/samtools/">http://sourceforge.net/projects/samtools/</a></p>
<p>IGV:<br />
<a href="http://www.broadinstitute.org/igv/">http://www.broadinstitute.org/igv/</a></p>
<p>Genome Analysis Toolkit:<br />
<a href="http://www.broadinstitute.org/gatk/">http://www.broadinstitute.org/gatk/</a></p>
<p>Picard:<br />
<a href="http://picard.sourceforge.net/">http://picard.sourceforge.net/</a></p>
<p>MuTect:<br />
<a href="http://www.broadinstitute.org/cancer/cga/mutect/">http://www.broadinstitute.org/cancer/cga/mutect/</a></p>
<p>Strelka:<br />
<a href="https://sites.google.com/site/strelkasomaticvariantcaller/">https://sites.google.com/site/strelkasomaticvariantcaller/</a></p>
<p>VarScan2:<br />
<a href="https://github.com/dkoboldt/varscan/">https://github.com/dkoboldt/varscan/</a></p>
<p>Variant Effect Predictor:<br />
<a href="http://www.ensembl.org/info/docs/tools/vep">http://www.ensembl.org/info/docs/tools/vep</a></p>
<p>GEMINI:<br />
<a href="http://gemini.readthedocs.org">http://gemini.readthedocs.org</a></p>
<h3 id="sources-of-data">Sources of Data<a class="headerlink" href="#sources-of-data" title="Permanent link">&para;</a></h3>
<p><a href="http://sra.dnanexus.com/studies/ERP001071">http://sra.dnanexus.com/studies/ERP001071</a><br />
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22194472">http://www.ncbi.nlm.nih.gov/pubmed/22194472</a></p>
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>This talk is based on Introduction to DNA-Seq processing for cancer data
by Mathieu Bourgey, Ph.D</p>
<p>*<a href="https://bitbucket.org/mugqic/muqpic">https://bitbucket.org/mugqic/muqpic</a>_pipelines*</p>
<p>The goal of this hands-on session is to present the main steps that are
commonly used to process and to analyze cancer sequencing data. We will
focus only on whole genome data and provide command lines that allow
detecting Single Nucleotide Variants (SNV). This workshop will show you
how to launch individual steps of a complete DNA-Seq SNV pipeline using
cancer data.</p>
<p>In the second part of the tutorial we will also be using IGV to
visualise and manually inspect candidate variant calls.</p>
<h2 id="prepare-the-environment">Prepare the Environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h2>
<p>We will use a dataset derived from whole genome sequencing of a
33-yr-old lung adenocarcinoma patient, who is a never-smoker and has no
familial cancer history.</p>
<p>The data consists of whole genome sequencing of liver metastatic lung
cancer (frozen), primary lung cancer (FFPE) and blood tissue of a lung
adenocarcinoma patient (AK55).</p>
<p>The BAM alignment files are contained in the subdirectory called
<code class="codehilite">alignment</code> and are located in the following subdirectories:</p>
<p><code class="codehilite">normal/normal.sorted.bam</code> and <code class="codehilite">normal/normal.sorted.bam.bai</code>
:   \</p>
<p><code class="codehilite">tumor/tumor.sorted.bam</code> and <code class="codehilite">tumor/tumor.sorted.bam.bai</code>
:   \</p>
<p>These files are based on subsetting the whole genomes derived from blood
and liver metastases to the first 10Mb of chromosome 4. This will allow
our analyses to run in a sufficient time during the workshop, but it’s
worth being aware that this is less \&lt; 0.5% of the genome which
highlights the length of time and resourced required to perform cancer
genomics on full genomes!</p>
<p>The initial structure of your folders should look like this:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>-- alignment/             # bam files
  -- normal/              # The blood sample directory containing bam files
  -- tumour/               # The tumour sample directory containing bam files \\
-- ref/                   # Contains reference genome files      
-- commands.sh            # cheat sheet
</pre></div>
</td></tr></table>

<p>Now we need to set some environment variables to save typing lengthy
file paths over and over. Copy and paste the following commands into
your terminal.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>export APP_ROOT=/home/trainee/snv/Applications

export IGVTOOLS_PATH=$PATH:$APP_ROOT/igvtools/

export PICARD_JAR=$APP_ROOT/picard/picard.jar

export SNPEFF_HOME=$APP_ROOT/snpeff/

export GATK_JAR=$APP_ROOT/gatk/GenomeAnalysisTK.jar

export BVATOOLS_JAR=$APP_ROOT/bvatools/bvatools-1.6-full.jar

export TRIMMOMATIC_JAR=$APP_ROOT/trimmomatic/trimmomatic-0.33.jar

export STRELKA_HOME=$APP_ROOT/strelka/

export MUTECT_JAR=$APP_ROOT/mutect/muTect-1.1.4.jar

export VARSCAN_JAR=$APP_ROOT/varscan/VarScan.v2.4.1.jar

export REF=/home/trainee/snv/ref

export SNV_BASE=/home/trainee/snv

export JAVA7=/usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java

export IGV=$APP_ROOT/igv/igv.sh
</pre></div>
</td></tr></table>

<p>Open the Terminal and go to the base <code class="codehilite">/</code>home/trainee/snv working
directory:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd $SNV_BASE
</pre></div>
</td></tr></table>

<p>All commands entered into the terminal for this tutorial should be from
within the <strong><code class="codehilite">/</code>home/trainee/snv</strong> directory.</p>
<p>Check that the <code class="codehilite">alignment</code> directory contains the above-mentioned files
by typing:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ls alignment
</pre></div>
</td></tr></table>

<h2 id="bam-files">BAM Files<a class="headerlink" href="#bam-files" title="Permanent link">&para;</a></h2>
<p>Let’s spend some time to explore bam files.</p>
<h3 id="step-1-exploring-bam-files">Step 1: Exploring BAM files<a class="headerlink" href="#step-1-exploring-bam-files" title="Permanent link">&para;</a></h3>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools view alignment/normal/normal.sorted.bam | head -n4
</pre></div>
</td></tr></table>

<p>Here you have examples of alignment results. A full description of the
flags can be found in the SAM specification
<a href="http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a></p>
<p>Another useful bit of information in the SAM is the CIGAR string. It’s
the 6<sup>th</sup> column in the file.</p>
<p>This column explains how the alignment was achieved.</p>
<p>M == base aligns <strong>but doesn’t have to be a match.</strong> A SNP will have an
M even if it disagrees with the reference.\
I == Insertion\
D == Deletion\
S == soft-clips. These are handy to find un removed adapters, viral
insertions, etc.</p>
<p>An in depth explanation of the CIGAR can be found
<a href="http://genome.sph.umich.edu/wiki/SAM">http://genome.sph.umich.edu/wiki/SAM</a></p>
<p>The exact details of the cigar string can be found in the SAM spec as
well.</p>
<p>We won’t go into too much detail at this point since we want to
concentrate on cancer specific issues now.</p>
<p>Now, you can try using picards explain flag site to understand what is
going on with your reads
<a href="http://broadinstitute.github.io/picard/explain-flags.html">http://broadinstitute.github.io/picard/explain-flags.html</a></p>
<p>There are 3 unique flags, what do they mean? The flag is the second
column.</p>
<p><strong>129:</strong>\
read paired\
second in pair\
\
<strong>113:</strong>\
read paired\
read reverse strand\
mate reverse strand\
first in pair\
\
<strong>161:</strong>\
read paired\
mate reverse strand\
second in pair\</p>
<p>There are lots of possible different flags, let’s look at a few more</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools view alignment/normal/normal.sorted.bam | head -n100
</pre></div>
</td></tr></table>

<p>Let’s take the last one, which looks properly paired and find it’s mate
pair.\
HINT: Instead of using ’head’ what unix command could we pipe the output
to? HINT2: Once we’ve found both reads to stop the command running we
type CTRL-C</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools view alignment/normal/normal.sorted.bam | grep HWI-ST478_0133:4:2205:14675:32513
</pre></div>
</td></tr></table>

<p>Using the cigar string, what can we tell about the alignment of the mate
pair?</p>
<p>The mate pair has a less convincing alignment with two insertions and
soft clipping reported.</p>
<p>How might the alignment information from the original read be used by
the aligner?</p>
<p>Even though the alignment of the mate pair is questionable the presence
of it’s properly paired mate helps the aligner in deciding where to put
the less-certain read.</p>
<p>You can use samtools to filter reads as well.</p>
<p>How many reads mapped and unmapped were there?\
HINT: Look at the samtools view help menu by typing samtools view
without any arguments</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools view -c -f4 alignment/normal/normal.sorted.bam
</pre></div>
</td></tr></table>

<p>77229</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools view -c -F4 alignment/normal/normal.sorted.bam
</pre></div>
</td></tr></table>

<p>22972373</p>
<h3 id="step-2-pre-processing-indel-realignment">Step 2: Pre-processing: Indel Realignment<a class="headerlink" href="#step-2-pre-processing-indel-realignment" title="Permanent link">&para;</a></h3>
<p>The first step for this is to realign around indels and snp dense
regions.\
The Genome Analysis toolkit has a tool for this called IndelRealigner.\
It basically runs in 2 steps:\
1. Find the targets\
2. Realign them\</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="nv">$JAVA7</span> -Xmx2G  -jar <span class="cp">${</span><span class="n">GATK_JAR</span><span class="cp">}</span> \
  -T RealignerTargetCreator \
  -R <span class="cp">${</span><span class="n">REF</span><span class="cp">}</span>/human_g1k_v37.fasta \
  -o alignment/normal/realign.intervals \
  -I alignment/normal/normal.sorted.bam \
  -I alignment/tumour/tumour.sorted.bam \
  -L <span class="cp">${</span><span class="n">REF</span><span class="cp">}</span>/human_g1k_v37.intervals

<span class="nv">$JAVA7</span> -Xmx2G -jar <span class="cp">${</span><span class="n">GATK_JAR</span><span class="cp">}</span> \
  -T IndelRealigner \
  -R <span class="cp">${</span><span class="n">REF</span><span class="cp">}</span>/human_g1k_v37.fasta \
  -targetIntervals alignment/normal/realign.intervals \
  --nWayOut .realigned.bam \
  -I alignment/normal/normal.sorted.bam \
  -I alignment/tumour/tumour.sorted.bam \
  -L <span class="cp">${</span><span class="n">REF</span><span class="cp">}</span>/human_g1k_v37.intervals

  mv normal.sorted.realigned.ba* alignment/normal/
  mv tumour.sorted.realigned.ba* alignment/tumour/
</pre></div>
</td></tr></table>

<p>Explanation of parameters</p>
<p>-I
:   BAM file(s)</p>
<p>-T
:   GATK algorithm to run</p>
<p>-R
:   the reference genome used for mapping (b37 from GATK here)</p>
<p>-jar
:   Path to GATK jar file</p>
<p>-L
:   Genomic intervals to operate on</p>
<p>Why did we use both normal and tumor together?</p>
<p>Because if a region needs realignment, maybe one of the sample in the
pair has less reads or was excluded from the target creation.\
This makes sure the normal and tumor are all in-sync for the somatic
calling step.</p>
<p>How many regions did it think needed cleaning ?</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>wc -l alignment/normal/realign.intervals -&gt; 27300
</pre></div>
</td></tr></table>

<p>Indel Realigner also makes sure the called deletions are left aligned
when there is a microsatellite or homopolymer.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8
9</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>This
ATCGAAAA-TCG
into
ATCG-AAAATCG

or
ATCGATATATATA--TCG
into
ATCG--ATATATATATCG
</pre></div>
</td></tr></table>

<p>Why it is important ?</p>
<p>This makes it easier for down stream analysis tools</p>
<p>For NGS analysis, the convention is to left align indels.</p>
<p>This is only really needed when calling variants with legacy locus-based
tools such as samtools or GATK UnifiedGenotyper. Otherwise you will have
worse performance and accuracy.</p>
<p>With more sophisticated tools (like GATK HaplotypeCaller) that involve
reconstructing haplotypes (eg through reassembly), the problem of
multiple valid representations is handled internally and does not need
to be corrected explicitly.</p>
<h3 id="step-3-pre-processing-fixmates">Step 3: Pre-processing: Fixmates<a class="headerlink" href="#step-3-pre-processing-fixmates" title="Permanent link">&para;</a></h3>
<p>Some read entries don’t have their mate information written properly.\
We use Picard to do this:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="nv">$JAVA7</span> -Xmx2G -jar <span class="cp">${</span><span class="n">PICARD_JAR</span><span class="cp">}</span> FixMateInformation \
  VALIDATION_STRINGENCY=SILENT \
  CREATE_INDEX=true \
  SORT_ORDER=coordinate \
  MAX_RECORDS_IN_RAM=500000 \
  INPUT=alignment/normal/normal.sorted.realigned.bam \
  OUTPUT=alignment/normal/normal.matefixed.bam

<span class="nv">$JAVA7</span> -Xmx2G -jar <span class="cp">${</span><span class="n">PICARD_JAR</span><span class="cp">}</span> FixMateInformation \
  VALIDATION_STRINGENCY=SILENT \
  CREATE_INDEX=true \
  SORT_ORDER=coordinate \
  MAX_RECORDS_IN_RAM=500000 \
  INPUT=alignment/tumour/tumour.sorted.realigned.bam \
  OUTPUT=alignment/tumour/tumour.matefixed.bam
</pre></div>
</td></tr></table>

<h3 id="step-4-pre-processing-mark-duplicates">Step 4: Pre-processing: Mark Duplicates<a class="headerlink" href="#step-4-pre-processing-mark-duplicates" title="Permanent link">&para;</a></h3>
<p>What are duplicate reads ?</p>
<p>Different read pairs representing the same initial DNA fragment.</p>
<p>What are they caused by ?</p>
<p>PCR reactions (PCR duplicates)\
Some clusters that are thought of being separate in the flowcell but are
the same (optical duplicates)</p>
<p>What are the ways to detect them ?</p>
<p>Picard and samtools uses the alignment positions:\
- Both 5’ ends of both reads need to have the same positions.\
- Each reads have to be on the same strand as well.\
\
Another method is to use a kmer approach:\
- take a part of both ends of the fragment\
- build a hash table\
- count the similar hits\
\
Brute force, compare all the sequences.</p>
<p>Here we will use picards approach:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="nv">$JAVA7</span> -Xmx2G -jar <span class="cp">${</span><span class="n">PICARD_JAR</span><span class="cp">}</span> MarkDuplicates \
  REMOVE_DUPLICATES=false \
  CREATE_MD5_FILE=true \
  VALIDATION_STRINGENCY=SILENT \
  CREATE_INDEX=true \
  INPUT=alignment/normal/normal.matefixed.bam \
  OUTPUT=alignment/normal/normal.sorted.dup.bam \
  METRICS_FILE=alignment/normal/normal.sorted.dup.metrics

<span class="nv">$JAVA7</span> -Xmx2G -jar <span class="cp">${</span><span class="n">PICARD_JAR</span><span class="cp">}</span> MarkDuplicates \
  REMOVE_DUPLICATES=false \
  CREATE_MD5_FILE=true \
  VALIDATION_STRINGENCY=SILENT \
  CREATE_INDEX=true \
  INPUT=alignment/tumour/tumour.matefixed.bam \
  OUTPUT=alignment/tumour/tumour.sorted.dup.bam \
  METRICS_FILE=alignment/tumour/tumour.sorted.dup.metrics
</pre></div>
</td></tr></table>

<p>We can look in the metrics output to see what happened.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less alignment/normal/normal.sorted.dup.metrics
</pre></div>
</td></tr></table>

<p>What percent of reads are duplicates?</p>
<p>0.046996%</p>
<p>Often, we have multiple libraries and when this occurs separate measures
are calculated for each library. Why is this important to do not combine
everything ?</p>
<p>Each library represents a set of different DNA fragments.</p>
<p>Each library involves different PCR reactions</p>
<p>So PCR duplicates can not occur between fragment of two different
libraries.</p>
<p>But similar fragment could be found between libraries when the coverage
is high.</p>
<h3 id="step-4-pre-processing-base-quality-recalibration">Step 4: Pre-processing: Base Quality Recalibration<a class="headerlink" href="#step-4-pre-processing-base-quality-recalibration" title="Permanent link">&para;</a></h3>
<p>Why do we need to recalibrate base quality scores ?</p>
<p>The vendors tend to inflate the values of the bases in the reads. The
recalibration tries to lower the scores of some biased motifs for some
technologies.</p>
<p>It runs in 2 steps,\
1- Build covariates based on context and known snp sites\
2- Correct the reads based on these metrics</p>
<p>GATK BaseRecalibrator:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">for</span> <span class="n">i</span> <span class="n">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="k">do</span>
  <span class="nv">$JAVA7</span> -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">GATK_JAR</span>} \
    -<span class="n">T</span> <span class="n">BaseRecalibrator</span> \
    -<span class="n">nct</span> <span class="mi">2</span> \
    -<span class="n">R</span> ${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
    -<span class="n">knownSites</span> ${<span class="n">REF</span>}/<span class="n">dbSnp-138_chr4</span>.<span class="n">vcf</span> \
    -<span class="n">L</span> <span class="mi">4</span>:<span class="mi">1</span><span class="o">-</span><span class="mi">10000000</span> \
    -<span class="n">o</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recalibration_report</span>.<span class="n">grp</span> \
    -<span class="n">I</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">bam</span>

    <span class="nv">$JAVA7</span> -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">GATK_JAR</span>} \
      -<span class="n">T</span> <span class="n">PrintReads</span> \
      -<span class="n">nct</span> <span class="mi">2</span> \
      -<span class="n">R</span> ${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
      -<span class="n">BQSR</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recalibration_report</span>.<span class="n">grp</span> \
      -<span class="n">o</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
      -<span class="n">I</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">bam</span>
<span class="n">done</span>
</pre></div>
</td></tr></table>

<h2 id="bam-qc">BAM QC<a class="headerlink" href="#bam-qc" title="Permanent link">&para;</a></h2>
<p>Once your whole bam is generated, it’s always a good thing to check the
data again to see if everything makes sense.</p>
<h3 id="step-1-bam-qc-compute-coverage">Step 1: BAM QC: Compute Coverage<a class="headerlink" href="#step-1-bam-qc-compute-coverage" title="Permanent link">&para;</a></h3>
<p>If you have data from a capture kit, you should see how well your
targets worked. Both GATK and BVATools have depth of coverage tools.\
Here we’ll use the GATK one</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">for</span> <span class="n">i</span> <span class="n">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="k">do</span>
  <span class="nv">$JAVA7</span>  -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">GATK_JAR</span>} \
    -<span class="n">T</span> <span class="n">DepthOfCoverage</span> \
    --<span class="n">omitDepthOutputAtEachBase</span> \
    --<span class="n">summaryCoverageThreshold</span> <span class="mi">10</span> \
    --<span class="n">summaryCoverageThreshold</span> <span class="mi">25</span> \
    --<span class="n">summaryCoverageThreshold</span> <span class="mi">50</span> \
    --<span class="n">summaryCoverageThreshold</span> <span class="mi">100</span> \
    --<span class="n">start</span> <span class="mi">1</span> --<span class="n">stop</span> <span class="mi">500</span> --<span class="n">nBins</span> <span class="mi">499</span> -<span class="n">dt</span> <span class="n">NONE</span> \
    -<span class="n">R</span> ${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
    -<span class="n">o</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">coverage</span> \
    -<span class="n">I</span> <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
    -<span class="n">L</span> <span class="mi">4</span>:<span class="mi">1</span><span class="o">-</span><span class="mi">10000000</span>
<span class="n">done</span>
</pre></div>
</td></tr></table>

<p>Explanation of parameters</p>
<p>omitBaseOutput
:   Do not output depth of coverage at each base</p>
<p>summaryCoverageThreshol
:   Coverage threshold (in percent) for summarizing statistics</p>
<p>dt
:   down sampling</p>
<p>L
:   Genomic intervals to operate on</p>
<p>Coverage is expected to be  25x in these project Look at the coverage:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less -S alignment/normal/normal.sorted.dup.recal.coverage.sample_interval_summary
less -S alignment/tumour/tumour.sorted.dup.recal.coverage.sample_interval_summary
</pre></div>
</td></tr></table>

<p>Is the coverage fit with the expectation ?</p>
<p>Yes the mean coverage of the region is 25x:</p>
<p>summaryCoverageThreshold is a useful function to see if your coverage is
uniform.</p>
<p>Another way is to compare the mean to the median. If both are quite
different that means something is wrong in your coverage.</p>
<h3 id="step-2-bam-qc-insert-size">Step 2: BAM QC: Insert Size<a class="headerlink" href="#step-2-bam-qc-insert-size" title="Permanent link">&para;</a></h3>
<p>It corresponds to the size of DNA fragments sequenced.</p>
<p>Different from the gap size (= distance between reads) !</p>
<p>These metrics are computed using Picard:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">for</span> <span class="n">i</span> <span class="n">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="k">do</span>
  <span class="nv">$JAVA7</span> -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">PICARD_JAR</span>} <span class="n">CollectInsertSizeMetrics</span> \
    <span class="n">VALIDATION_STRINGENCY</span>=<span class="n">SILENT</span> \
    <span class="n">REFERENCE_SEQUENCE</span>=${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
    <span class="n">INPUT</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
    <span class="n">OUTPUT</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">metric</span>.<span class="n">insertSize</span>.<span class="n">tsv</span> \
    <span class="n">HISTOGRAM_FILE</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">metric</span>.<span class="n">insertSize</span>.<span class="n">histo</span>.<span class="n">pdf</span> \
    <span class="n">METRIC_ACCUMULATION_LEVEL</span>=<span class="n">LIBRARY</span>
<span class="n">done</span>
</pre></div>
</td></tr></table>

<p>look at the output</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less -S alignment/normal/normal.sorted.dup.recal.metric.insertSize.tsv
less -S alignment/tumour/tumour.sorted.dup.recal.metric.insertSize.tsv
</pre></div>
</td></tr></table>

<p>How do the two libraries compares?</p>
<p>The tumour sample has a larger median insert size than the normal sample
(405 vs 329)</p>
<h3 id="step-3-bam-qc-alignment-metrics">Step 3: BAM QC: Alignment metrics<a class="headerlink" href="#step-3-bam-qc-alignment-metrics" title="Permanent link">&para;</a></h3>
<p>It tells you if your sample and you reference fit together</p>
<p>For the alignment metrics, samtools flagstat is very fast but with
bwa-mem since some reads get broken into pieces, the numbers are a bit
confusing.</p>
<p>We prefer the Picard way of computing metrics:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8
9</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">for</span> <span class="n">i</span> <span class="n">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="k">do</span>
  <span class="nv">$JAVA7</span> -<span class="n">Xmx2G</span> -<span class="n">jar</span> ${<span class="n">PICARD_JAR</span>} <span class="n">CollectAlignmentSummaryMetrics</span> \
    <span class="n">VALIDATION_STRINGENCY</span>=<span class="n">SILENT</span> \
    <span class="n">REFERENCE_SEQUENCE</span>=${<span class="n">REF</span>}/<span class="n">human_g1k_v37</span>.<span class="n">fasta</span> \
    <span class="n">INPUT</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
    <span class="n">OUTPUT</span>=<span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">metric</span>.<span class="n">alignment</span>.<span class="n">tsv</span> \
    <span class="n">METRIC_ACCUMULATION_LEVEL</span>=<span class="n">LIBRARY</span>
<span class="n">done</span>
</pre></div>
</td></tr></table>

<p>explore the results</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less -S alignment/normal/normal.sorted.dup.recal.metric.alignment.tsv
less -S alignment/tumour/tumour.sorted.dup.recal.metric.alignment.tsv
</pre></div>
</td></tr></table>

<p>Do you think the sample and the reference genome fit together ?</p>
<p>Yes, 99% of the reads have been aligned\
Usually, we consider:\
- A good alignment if > 85%\
- Reference assembly issues if [60-85]%\
- Probably a mismatch between sample and ref if \&lt; 60 %</p>
<h2 id="variant-calling">Variant Calling<a class="headerlink" href="#variant-calling" title="Permanent link">&para;</a></h2>
<p>Most of SNV caller use either a Baysian, a threshold or a t-test
approach to do the calling</p>
<p>Here we will try 3 variant callers.\
- Varscan 2\
- MuTecT\
- Strelka</p>
<p>Other candidates\
- Virmid\
- Somatic sniper</p>
<p>many, MANY others can be found here: <a href="https://www.biostars.org/p/19104/">https://www.biostars.org/p/19104/</a></p>
<p>In our case, let’s create a new work directory to start with (from base
directory):</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd $SNV_BASE
mkdir variant_calling
</pre></div>
</td></tr></table>

<p>varscan 2 VarScan calls somatic variants (SNPs and indels) using a
heuristic method and a statistical test based on the number of aligned
reads supporting each allele.</p>
<p>Varscan somatic caller expects both a normal and a tumour file in
SAMtools pileup format from sequence alignments in binary alignment/map
(BAM) format. To build a pileup file, you will need:</p>
<ul>
<li>
<p>A SAM/BAM file (&ldquo;myData.bam&rdquo;) that has been sorted using the sort
command of SAMtools. - The reference sequence (&ldquo;reference.fasta&rdquo;) to
which reads were aligned, in FASTA format. - The SAMtools software
package.</p>
<p>for i in normal tumour
do
samtools mpileup -L 1000 -B -q 1 \
  -f ${REF}/human_g1k_v37.fasta \
  -r 4:1-10000000 \
  alignment/${i}/${i}.sorted.dup.recal.bam \</p>
<blockquote>
<p>variant_calling/${i}.mpileup
done</p>
</blockquote>
<p>$JAVA7 -Xmx2G -jar ${VARSCAN_JAR} \
somatic variant_calling/normal.mpileup \
variant_calling/tumour.mpileup \
variant_calling/varscan \
&ndash;output-vcf 1 \
&ndash;strand-filter 1 \
&ndash;somatic-p-value 0.001</p>
</li>
</ul>
<p>Notes on samtools arguments</p>
<p>-L
:   = max per-sample depth for INDEL calling [1000] ;</p>
<p>-B
:   = disable BAQ (per-Base Alignment Quality) ;</p>
<p>-q
:   = skip alignments with mapQ smaller than 1 ;</p>
<p>-g
:   = generate genotype likelihoods in BCF format</p>
<p>Notes on bcftools arguments</p>
<p>-v
:   = output potential variant sites only</p>
<p>-c
:   = SNP calling (force –e : likelihood based analyses)</p>
<p>-g
:   = call genotypes at variant sites</p>
<p>Now let’s try a different variant caller, MuTect\</p>
<p>Note MuTecT only works with Java 6, 7 will give you an error\
if you get &ldquo;Comparison method violates its general contract!\
you used java 7&rdquo;</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>java -Xmx2G -jar <span class="cp">${</span><span class="n">MUTECT_JAR</span><span class="cp">}</span> \
  -T MuTect \
  -R <span class="cp">${</span><span class="n">REF</span><span class="cp">}</span>/human_g1k_v37.fasta \
  -dt NONE -baq OFF --validation_strictness LENIENT -nt 2 \
  --dbsnp <span class="cp">${</span><span class="n">REF</span><span class="cp">}</span>/dbSnp-138_chr4.vcf \
  --input_file:normal alignment/normal/normal.sorted.dup.recal.bam \
  --input_file:tumor alignment/tumour/tumour.sorted.dup.recal.bam \
  --out variant_calling/mutect.call_stats.txt \
  --coverage_file variant_calling/mutect.wig.txt \
  -pow variant_calling/mutect.power \
  -vcf variant_calling/mutect.vcf \
  -L 4:1-10000000
</pre></div>
</td></tr></table>

<p>And finally let’s try Illumina’s Strelka</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cp <span class="cp">${</span><span class="n">STRELKA_HOME</span><span class="cp">}</span>/etc/strelka_config_bwa_default.ini .

sed &#39;s/isSkipDepthFilters =.*/isSkipDepthFilters = 1/g&#39; -i strelka_config_bwa_default.ini

<span class="cp">${</span><span class="n">STRELKA_HOME</span><span class="cp">}</span>/bin/configureStrelkaWorkflow.pl \
  --normal=alignment/normal/normal.sorted.dup.recal.bam \
  --tumor=alignment/tumour/tumour.sorted.dup.recal.bam \
  --ref=<span class="cp">${</span><span class="n">REF</span><span class="cp">}</span>/human_g1k_v37.fasta \
  --config=<span class="cp">${</span><span class="n">SNV_BASE</span><span class="cp">}</span>/strelka_config_bwa_default.ini \
  --output-dir=variant_calling/strelka/

  cd variant_calling/strelka/
  make -j2
  cd ../..

  cp variant_calling/strelka/results/passed.somatic.snvs.vcf variant_calling/strelka.vcf
</pre></div>
</td></tr></table>

<p>Now we have variants from all three methods. Let’s compress and index
the vcfs for future visualisation.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>for i in variant_calling/*.vcf;do bgzip -c $i &gt; $i.gz ; tabix -p vcf $i.gz;done
</pre></div>
</td></tr></table>

<p>Let’s look at a compressed vcf.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>zless -S variant_calling/varscan.snp.vcf.gz
</pre></div>
</td></tr></table>

<p>Details on the spec can be found here:
<a href="http://vcftools.sourceforge.net/specs.html">http://vcftools.sourceforge.net/specs.html</a></p>
<p>Fields vary from caller to caller.</p>
<p>Some values are are almost always there:\
- The ref vs alt alleles,\
- variant quality (QUAL column)\
- The per-sample genotype (GT) values.</p>
<p>Note on vcf fields</p>
<p>DP
:   = &ldquo;Raw read depth&rdquo;</p>
<p>GT
:   = &ldquo;Genotype&rdquo;</p>
<p>PL
:   = &ldquo;List of Phred-scaled genotype likelihoods&rdquo; (min is better)</p>
<p>DP
:   = &ldquo;# high-quality bases&rdquo;</p>
<p>SP
:   = &ldquo;Phred-scaled strand bias P-value&rdquo;</p>
<p>GQ
:   = &ldquo;Genotype Quality&rdquo;</p>
<p>Looking at the three vcf files, how can we detect only somatic variants?</p>
<p>some commands to find somatic variant in the vcf file</p>
<p>varscan</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>grep SOMATIC variant_calling/varscan.snp.vcf
</pre></div>
</td></tr></table>

<p>MuTecT</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>grep -v REJECT variant_calling/mutect.vcf | grep -v &quot;^#&quot;
</pre></div>
</td></tr></table>

<p>Strelka</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>grep -v &quot;^#&quot; variant_calling/strelka.vcf
</pre></div>
</td></tr></table>

<h2 id="variant-visualisation">Variant Visualisation<a class="headerlink" href="#variant-visualisation" title="Permanent link">&para;</a></h2>
<p>The Integrative Genomics Viewer (IGV) is an efficient visualization tool
for interactive exploration of large genome datasets.</p>
<p>Before jumping into IGV, we’ll generate a track IGV can use to plot
coverage:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">for</span> <span class="n">i</span> <span class="n">in</span> <span class="n">normal</span> <span class="n">tumour</span>
<span class="k">do</span>
  <span class="nv">$JAVA7</span> -<span class="n">jar</span> ${<span class="n">IGVTOOLS_PATH</span>}/<span class="n">igvtools</span>.<span class="n">jar</span> <span class="nb">count</span> \
    -<span class="n">f</span> <span class="nb">min</span>,<span class="nb">max</span>,<span class="n">mean</span> \
    <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span> \
    <span class="n">alignment</span>/${<span class="n">i</span>}/${<span class="n">i</span>}.<span class="n">sorted</span>.<span class="n">dup</span>.<span class="n">recal</span>.<span class="n">bam</span>.<span class="n">tdf</span> \
    <span class="n">b37</span>
<span class="n">done</span>
</pre></div>
</td></tr></table>

<p>Open IGV</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>$IGV
</pre></div>
</td></tr></table>

<p>Then: 1. Chose the reference genome corresponding to those use for
alignment (b37)\
2. Load bam files (tumour.sorted.dup.recal.bam and
normal.sorted.dup.recal.bam)\
3. Load vcf files (from variant_Calling directory)</p>
<p>Explore/play with the data:\
-find germline variants\
-find somatic variants\
-Look around&hellip;</p>
<h2 id="variant-annotation">Variant Annotation<a class="headerlink" href="#variant-annotation" title="Permanent link">&para;</a></h2>
<p>Following variant calling, we end up with a VCF file of genomic
coordinates with the genotype(s) and quality information for each
variant. By itself, this information is not much use to us unless there
is a specific genomic location we are interested in. Generally, we next
want to annotate these variants to determine whether they impact any
genes and if so what is their level of impact (e.g. are they causing a
premature stop codon gain or are they likely less harmful missense
mutations).</p>
<p>The sections above have dealt with calling somatic variants from the
first 10Mb of chromosome 4. This is important in finding variants that
are unique to the tumour sample(s) and may have driven both tumour
growth and/or metastasis. An important secondary question is whether the
germline genome of the patient contains any variants that may have
contributed to the development of the initial tumour through
predisposing the patient to cancer. These variants <em>may not</em> be captured
by somatic variant analysis as their allele frequency may not change in
the tumour genome compared with the normal.</p>
<p>For this section, we will use <strong>all</strong> variants from the first 60Mb of
chromosome 5 that have been pre-generated using the GATK HaplotypeCaller
variant caller on both the normal and tumour genomes. The output of this
was GVCF files which were fed into GATK GenotypeGVCFs to produce a
merged VCF file. We will use this pre-generated file as we are primarily
interested in the annotation of variants rather than their generation.
The annotation method we will use is called <strong>Variant Effect Predictor</strong>
or VEP for short and is available from Ensembl here:
<a href="http://ensembl.org/info/docs/tools/vep/index.html">http://ensembl.org/info/docs/tools/vep/index.html</a>.</p>
<p>Our pre-generated VCF file is located in the <code class="codehilite">variants</code> folder. Let’s
have a quick look at the variants:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>zless variants/HC.chr5.60Mb.vcf.gz
</pre></div>
</td></tr></table>

<p>Notice how there are two genotype blocks at the end of each line for the
normal (Blood) and tumour (liverMets) samples.</p>
<p>Let’s now run VEP on this VCF file to annotate each variant with its
impact(s) on the genome.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>perl Applications/ensembl-tools/scripts/variant_effect_predictor/variant_effect_predictor.pl -i variants/HC.chr5.60Mb.vcf.gz --vcf -o variants/HC.chr5.60Mb.vep.vcf --stats_file variants/HC.chr5.60Mb.vep.html --format vcf --offline -fork 4 --fasta ref/human_g1k_v37.fasta --fields Consequence,Codons,Amino_acids,Gene,SYMBOL,Feature,EXON,PolyPhen,SIFT,Protein_position,BIOTYPE
</pre></div>
</td></tr></table>

<p>VEP will take approximately 10 minutes to run and once it is finished
you will have a new VCF file with all of the information in the input
file but with added annotations in the INFO block. VEP also produces an
HTML report summarising the distribution and impact of variants
identified.</p>
<p>Once VEP is done running, let’s first look at the HTML report it
produced with the following command:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>firefox variants/HC.chr5.60Mb.vep.html
</pre></div>
</td></tr></table>

<p>This report shows information on the VEP run, the number of variants,
the classes of variants detected, the variant consequences and the
distributions of variants through the genome. Close Firefox to resume
the terminal prompt.</p>
<p>Now let’s look at the variant annotations that VEP has added to the VCF
file by focussing on a single variant. Let’s fetch the same variant from
the original VCF file and the annotated VCF file to see what has been
changed.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>zcat variants/HC.chr5.60Mb.vcf.gz | grep &#39;5\s174106\s&#39;
grep &#39;5\s174106\s&#39; variants/HC.chr5.60Mb.vep.vcf
</pre></div>
</td></tr></table>

<p>These commands give us the original variant:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>5   174106  .   G   A   225.44  .   AC=2;AF=0.500;AN=4;BaseQRankSum=1.22;ClippingRankSum=0.811;DP=21;FS=0.000;GQ_MEAN=127.00;GQ_STDDEV=62.23;MLEAC=2;MLEAF=0.500;MQ=60.00;MQ0=0;MQRankSum=0.322;NCC=0;QD=10.74;ReadPosRankSum=0.377;SOR=0.446   GT:AD:DP:GQ:PL  0/1:7,6:13:99:171,0,208 0/1:5,3:8:83:83,0,145
</pre></div>
</td></tr></table>

<p>and the same variant annotated is:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>5   174106  .   G   A   225.44  .   AC=2;AF=0.500;AN=4;BaseQRankSum=1.22;ClippingRankSum=0.811;DP=21;FS=0.000;GQ_MEAN=127.00;GQ_STDDEV=62.23;MLEAC=2;MLEAF=0.500;MQ=60.00;MQ0=0;MQRankSum=0.322;NCC=0;QD=10.74;ReadPosRankSum=0.377;SOR=0.446;CSQ=missense_variant|cGg/cAg|R/Q|ENSG00000153404|PLEKHG4B|ENST00000283426|16/18|||1076|protein_coding,non_coding_transcript_exon_variant&amp;non_coding_transcript_variant|||ENSG00000153404|PLEKHG4B|ENST00000504041|5/8||||retained_intron  GT:AD:DP:GQ:PL  0/1:7,6:13:99:171,0,208 0/1:5,3:8:83:83,0,145
</pre></div>
</td></tr></table>

<p>You can see that VEP has added:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>CSQ=missense_variant|cGg/cAg|R/Q|ENSG00000153404|PLEKHG4B|ENST00000283426|16/18|||1076|protein_coding,non_coding_transcript_exon_variant&amp;non_coding_transcript_variant|||ENSG00000153404|PLEKHG4B|ENST00000504041|5/8||||retained_intron
</pre></div>
</td></tr></table>

<p>This is further composed of two annotations for this variant:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>missense_variant|cGg/cAg|R/Q|ENSG00000153404|PLEKHG4B|ENST00000283426|16/18|||1076|protein_coding
</pre></div>
</td></tr></table>

<p>and</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>non_coding_transcript_exon_variant&amp;non_coding_transcript_variant|||ENSG00000153404|PLEKHG4B|ENST00000504041|5/8||||retained_intron
</pre></div>
</td></tr></table>

<p>The first of these is saying that this variant is a missense variant in
the gene PLEKHG4B for the transcript ENST00000283426 and the second that
it is also a non_coding_transcript_exon_variant in the transcript
ENST00000504041.</p>
<h2 id="variant-filtration">Variant Filtration<a class="headerlink" href="#variant-filtration" title="Permanent link">&para;</a></h2>
<p>We now have a VCF file where each variant has been annotated with one or
more impacts for one or more genes. In a typical whole cancer genome,
you will have about 4-5 million variants, and therefore rows, in a VCF
file which takes up gigabytes of space. In our small example, we have
just 100,000 variants which is already too large to make any kind of
meaningful sense out of by just opening up the VCF file in a text
editor. We need a solution that allows us to perform intelligent queries
on our variants to search this mass of noise for the signal we are
interested in.</p>
<p>Luckily, such a free tool exists and is called GEMINI. GEMINI takes as
an input your annotated VCF file and creates a database file which it
can then query using Structured Query Language (SQL) commands. Not only
does GEMINI make your variants easily searchable, it also brings in many
external annotations to add more information about your variants (such
as their frequencies in international databases).</p>
<p>To get started with GEMINI, let’s make a database out of our annotated
VCF file.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gemini load -v variants/HC.chr5.60Mb.vep.vcf --cores 4 --skip-gerp-bp --skip-cadd -t VEP variants/HC.chr5.60Mb.vep.vcf.db
</pre></div>
</td></tr></table>

<p>This will take approximately 10 minutes. You will see a few errors due
to multiallelic sites, normally these sites are decomposed and
normalized before creating the GEMINI database but this is outside the
scope of this workshop.</p>
<p>Once the database has been created let’s run a basic query to see what
kind of information we get out of GEMINI.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gemini query -q &quot;SELECT *, (gts).(*), (gt_types).(*), (gt_depths).(*), (gt_ref_depths).(*), (gt_alt_depths).(*), (gt_quals).(*) FROM variants LIMIT 10;&quot; --header variants/HC.chr5.60Mb.vep.vcf.db
</pre></div>
</td></tr></table>

<p>This will output a bunch of ordered information for your query to the
command line, this is usually saved to a TSV file and opened in a
spreadsheet as we will do for the next query. In the mean time, let’s
dissect this query to understand the syntax we need to use to filter our
variants. First, we have a SELECT statement which simply specifies that
we want to select data from the database. The following comma-separated
values are the columns that we want to output from the database, in this
case we are selecting all columns with the star character and then all
sub-columns for each sample with the other values. Then, we have a &ldquo;FROM
variants&rdquo; statement which is specifying the table within the database
that we want to fetch information from. Finally, the &ldquo;LIMIT 10&rdquo;
statement specifies that no more than 10 rows should be returned. In
summary then, we are asking for all columns for 10 rows from the table
<code class="codehilite">variants</code>. If you haven’t used SQL before don’t worry, the GEMINI
website is very helpful and provides many examples for how to query your
database.</p>
<p>Let’s now perform a more interesting query to find variants that have a
medium or high impact on a gene and are rare or not present in existing
international allele frequency databases. We will save the output of
this query to a file and open it up in a spreadsheet.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gemini query -q &quot;SELECT *, (gts).(*), (gt_types).(*), (gt_depths).(*), (gt_ref_depths).(*), (gt_alt_depths).(*), (gt_quals).(*) FROM variants WHERE (impact_severity = &#39;HIGH&#39; OR impact_severity = &#39;MED&#39;) AND (aaf_1kg_all &lt; 0.01 OR aaf_1kg_all is null) AND (aaf_esp_all &lt; 0.01 OR aaf_esp_all is null) AND (aaf_exac_all &lt; 0.01 OR aaf_exac_all is null);&quot; --header variants/HC.chr5.60Mb.vep.vcf.db &gt; variants/gemini-result.tsv
</pre></div>
</td></tr></table>

<p>Notice that we have added a WHERE statement which restricts the rows
that are returned based on values that we specify for specific columns.
Here, we are asking to return variants where their impact on the gene
(impact_severity column) is medium or high and the allele frequency in
1000Genomes, ESP and EXaC is less than 1% or the variant is not present
in any of these databases.</p>
<p>Now let’s open the result in a spreadsheet to look at the annotations:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>libreoffice --calc variants/gemini-result.tsv
Tick the &quot;Tab&quot; under &quot;Separated by&quot; on the dialog window that comes up.
</pre></div>
</td></tr></table>

<p>You can see that the first 14 columns contain information on the variant
including its location, ref, alt, dbSNP ID, quality and type. Slowly
scroll to the right and look at the columns of data that are provided.
Most importantly, column BD includes the gene this variant impacts, BN
the impact itself and BP the impact severity. Scroll towards the end of
the spreadsheet until you get to columns ED and EE, these contain the
genotype for each of the samples. Columns EH and EI contain the total
depth for each variant in each sample and the 4 following columns
contain the reference and alternate depths for each sample. Finally,
columns EN and EO contain the genotype qualities (from the GQ field in
the VCF) for each sample. As you scroll back and forth through this
spreadsheet, you will see that GEMINI brings in information from a
variety of sources including: OMIM, ClinVar, GERP, PolyPhen 2, SIFT,
ESP, 1000 Genomes, ExAC, ENCODE, CADD and more! We are only looking at a
small number of variants from the start of a chromosome so not many of
these annotations will be present but in a full genome database they are
incredibly useful.</p>
<p>GEMINI allows you to filter your variants based on any column that you
see in this results file. For example, you may want all variants in a
specific gene, in which case you would simply add &ldquo;WHERE gene = ’BRCA1’&rdquo;
to your query. For complete documentation with many examples of queries,
see the GEMINI documentation here: <a href="http://gemini.readthedocs.org">http://gemini.readthedocs.org</a>.</p>
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ol>
<li>
<p>Paila U, Chapman BA, Kirchner R and Quinlan AR. &ldquo;GEMINI: Integrative
    Exploration of Genetic Variation and Genome Annotations&rdquo;. PLoS
    Comput Biol, 2013, 9(7): e1003153. doi:10.1371/journal.pcbi.1003153</p>
</li>
<li>
<p>McLaren W, Pritchard B, Rios D, Chen Y, Flicek P and Cunningham F.
    &ldquo;Deriving the consequences of genomic variants with the Ensembl API
    and SNP Effect Predictor&rdquo;. Bioinformatics, 2010, 26(16):2069-70,
    doi:10.1093/bioinformatics/btq330</p>
</li>
</ol>
<h2 id="acknowledgements">Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permanent link">&para;</a></h2>
<p>This tutorial is an adaptation of the one created by Louis letourneau
<a href="https://github.com/lletourn/Workshops/tree/ebiCancerWorkshop201407doc/01-SNVCalling.md">https://github.com/lletourn/Workshops/tree/ebiCancerWorkshop201407doc/01-SNVCalling.md</a>.
I would like to thank and acknowledge Louis for this help and for
sharing his material. The format of the tutorial has been inspired from
Mar Gonzalez Porta. I also want to acknowledge Joel Fillon, Louis
Letrouneau (again), Francois Lefebvre, Maxime Caron and Guillaume
Bourque for the help in building these pipelines and working with all
the various datasets.</p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Read Alignment
              </span>
            </div>
          </a>
        
        
          <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Copy Number Variation
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-6b4ff16fa7.js"></script>
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/spoilers.js"></script>
      
    
    
      
    
  </body>
</html>